app/org/sagebionetworks/bridge/play/controllers/SurveyController.java

```diff
@@ -69,42 +70,15 @@ public Result getAllSurveysMostRecentlyPublishedVersion() throws Exception {
         return okResult(surveys);
     }
     
-    public Result getSurveyForUser(final String surveyGuid, final String createdOnString) throws Exception {
-        final UserSession session = getAuthenticatedAndConsentedSession();
-
-        ViewCacheKey<Survey> cacheKey = viewCache.getCacheKey(Survey.class, surveyGuid, createdOnString); 
-        String json = viewCache.getView(cacheKey, new Supplier<Survey>() {
-            @Override public Survey get() {
-                StudyIdentifier studyId = session.getStudyIdentifier();
-                long createdOn = DateUtils.convertToMillisFromEpoch(createdOnString);
-                GuidCreatedOnVersionHolder keys = new GuidCreatedOnVersionHolderImpl(surveyGuid, createdOn);
-
-                Survey survey = surveyService.getSurvey(keys);
-                verifySurveyIsInStudy(session, studyId, survey);
-                return surveyService.getSurvey(keys);
-            }
-        });
-        return ok(json).as(JSON_MIME_TYPE);
-    }
-
-    public Result getSurveyMostRecentlyPublishedVersionForUser(final String surveyGuid) throws Exception {
-        final UserSession session = getAuthenticatedAndConsentedSession();
-        
-        ViewCacheKey<Survey> cacheKey = viewCache.getCacheKey(Survey.class, surveyGuid, PUBLISHED_KEY);
-        String json = viewCache.getView(cacheKey, new Supplier<Survey>() {
-            @Override public Survey get() {
-                StudyIdentifier studyId = session.getStudyIdentifier();
-                Survey survey = surveyService.getSurveyMostRecentlyPublishedVersion(studyId, surveyGuid);
-                verifySurveyIsInStudy(session, studyId, survey);
-                return survey;
-            }
-        });
-        return ok(json).as(JSON_MIME_TYPE);
-    }
-    
     // Otherwise you don't need consent but you must be a researcher or an administrator
     public Result getSurvey(final String surveyGuid, final String createdOnString) throws Exception {
-        final UserSession session = getAuthenticatedSession(DEVELOPER);
+        final UserSession session = getSessionIfItExists();
+        if (session == null) {
+            throw new NotAuthenticatedException();
+        }
```

Comment:
getAuthenticatedSession() (without args) does the same thing as getSessionIfItExists(), except it also throws NotAuthenticatedException() if the session doesn't exist. This will save you having to write the same null check everywhere.